### PR TITLE
[WIP] Custom Resource Definition feature to replace Third Party Resources

### PR DESCRIPTION
**Pull request by kenan435**
Since Third Party Resources will be deprecated in k8s 1.8, we are introducing Custom Resource Definitions feature which is supposed to replace Third Party Resources. The two feature TPR and CRD will both be present on the dashboard during the k8s's 1.7 iteration to easy with migration from TPR's to CRD's

### CODE REVIEW

File: src/app/backend/api/types.go [hunk 1]
```diff
@@ -157,28 +159,29 @@ var KindToAPIMapping = map[string]struct {
...
+	ResourceKindReplicaSet:               {"replicasets", ClientTypeExtensionClient, true},
+	ResourceKindReplicationController:    {"replicationcontrollers", ClientTypeDefault, true},
+	ResourceKindResourceQuota:            {"resourcequotas", ClientTypeDefault, true},
+	ResourceKindSecret:                   {"secrets", ClientTypeDefault, true},
+	ResourceKindService:                  {"services", ClientTypeDefault, true},
+	ResourceKindStatefulSet:              {"statefulsets", ClientTypeAppsClient, true},
+	ResourceKindThirdPartyResource:       {"thirdpartyresources", ClientTypeExtensionClient, true},
+	ResourceKindCustomResourceDefinition: {"CustomResourceDefinition", ClientTypeAPIExtensionsClient, true},
+	ResourceKindStorageClass:             {"storageclasses", ClientTypeStorageClient, false},
+	ResourceKindEndpoint:                 {"endpoints", ClientTypeDefault, true},
```
Comment (maciaszczykm) [inline src/app/backend/api/types.go]: Be careful during merge if cron jobs will go to master first. It will not merge automatically due to indent change.

**Comment by kenan435** [comment]
Will have a look.

**Comment by maciaszczykm** [comment]
Should be done after Angular migration in the new branch.

/close

**Comment by kshafiee** [comment]
Any ideas if this work is already complete? Does the dashboard support CRDs today?

**Comment by shu-mutou** [comment]
@kshafiee please refer #2493

**Pull request closed without merging**